src/F2_autoreport.py

```diff
@@ -530,6 +530,39 @@ def create_app(results_json):
             message_string = substrings[-1]
             errors_dict.update({i: message_string})
 
+    # Build a pandas dataframe with the data for the various demands
+
+    # The below dict will gather all the keys of the various plots for later use in the graphOptions.csv
+    dict_for_plots = {"demands": {}, "supplies": {}}
+
+    # demands is a dict with all the info for the demands
+    # dem_keys is a list of keys of the demands dict
+
+    # Initialize an empty pandas dataframe which will store the demands and values
+    # Later, this dataframe can be passed to a function directly make the graphs with Plotly
+    df_all_demands = pd.DataFrame()
+
+    # variable to keep count of the columns in the pandas dataframe
+    cols_in_df_all_demands = 0
+
+    # The below loop will add all the demands to the dict_for_plots dictionary, including the timeseries values
+    for demand in dem_keys:
+        dict_for_plots["demands"].update(
+            {demand: results_json[ENERGY_CONSUMPTION][demand]["timeseries"]}
+        )
+        df_all_demands.insert(
```

Comment:
You can create the same dataframe with `df_all_demand = pd.DataFrame.from_dict(dict_for_plots["demands"], orient="columns")`

Reply:
damn, Pandas has so many ways of doing the same thing!

Thank you again. Replaced the .insert method with your line of code.